subprojects/docs/src/samples/build-organization/incubating/publishing-convention-plugins/groovy/convention-plugins/build.gradle

```diff
@@ -0,0 +1,50 @@
+// tag::apply[]
+// tag::publish[]
+plugins {
+    id 'groovy-gradle-plugin'
+// end::apply[]
+    id 'maven-publish'
+    id 'java'
+// tag::apply[]
+}
+// end::apply[]
+
+group = 'com.myorg.conventions'
+version = '1.0'
+
+publishing {
+    repositories {
+        maven {
+            // change to point to your repo, e.g. http://my.org/repo
+            url = layout.buildDirectory.dir("repo")
+        }
+    }
+}
+
+tasks.named('publish') {
+    dependsOn('check')
+}
+
+// end::publish[]
+
+testing {
+    suites {
+        test {
+            useJUnitJupiter()
```

Comment:
This isn't really correct because this causes us to add JUnit Jupiter dependencies as well as the Spock dependencies below. 

I think it's probably easiest for us to go ahead and add a `useSpock(...)` convenience method. Let's leave this as-is and do that in a separate change.

Alternatively, we could remove `useJUnitJupiter` and configure `useJUnitPlatform` directly on the task.